src/util/sandbox.cpp

```diff
@@ -369,80 +370,110 @@ QString Sandbox::migrateOldSettings() {
         return sandboxedPath;
     }
 
+    // Sandbox::askForAccess cannot be used here because it depends on settings being
+    // initialized. There is no need to store the bookmark anyway because this is a
+    // one time process.
+#ifdef __APPLE__
+
     // Because Mixxx cannot test if the old path exists before getting permission to access it
     // outside the sandbox, unfortunately it is necessary to annoy the user with this popup
     // even if they are installing Mixxx >= 2.3.0 without having installed an old version of Mixxx.
     QString title = QObject::tr("Upgrading old Mixxx settings");
-    QString oldPath = homePath + QLatin1String("/Library/Application Support/Mixxx");
     QMessageBox::information(nullptr,
             title,
             QObject::tr(
-                    "Due to Mac sandboxing, Mixxx needs your permission to "
-                    "access your music library "
-                    "and settings from Mixxx versions before 2.3.0. After "
-                    "clicking OK, you will see a file picker. "
-                    "To give Mixxx permission, press the Ok button in the file "
-                    "picker."
+                    "Existing Mixxx settings directory must be moved to new location."
                     "\n\n"
-                    "If you do not want to grant Mixxx access click Cancel on "
-                    "the file picker."));
+                    "Due to macOS sandboxing, Mixxx needs your permission to access your "
+                    "music library and move settings to the sandbox from Mixxx versions "
+                    "before 2.3.0. After clicking OK, you will see standard file open dialog. "
+                    "\n\n"
+                    "Select Open button in the file open dialog to grant Mixxx access to old files."
+                    "\n\n"
+                    "If you do not want to grant Mixxx access click Cancel on the file picker."));
     QString result = QFileDialog::getExistingDirectory(
-            nullptr,
-            title,
-            oldPath);
-    if (result != oldPath) {
-        qInfo() << "Sandbox::migrateOldSettings: User declined to migrate "
-                   "old settings from"
-                << oldPath << "User selected" << result;
+        nullptr,
+        title,
+        legacySettingsPath);
+    if (result != legacySettingsPath) {
+        qInfo() << "Sandbox::migrateOldSettings: User declined to migrate old settings from"
+                << legacySettingsPath << "User selected" << result;
         return sandboxedPath;
     }
 
-    // Sandbox::askForAccess cannot be used here because it depends on settings being
-    // initialized. There is no need to store the bookmark anyway because this is a
-    // one time process.
-#ifdef __APPLE__
+    // Check if Mixxx process is sandboxed
+    QString sandboxContainerId = QProcessEnvironment::systemEnvironment().value("APP_SANDBOX_CONTAINER_ID", "");
```

Comment:
According to this [StackOverflow comment](https://stackoverflow.com/questions/12177948/how-do-i-detect-if-my-app-is-sandboxed#comment64321443_12178610) this doesn't work as of macOS 10.12, but it works locally for you?